databuilder/query_language.py

```diff
@@ -328,6 +328,12 @@ def is_between(self, start, end):
     def is_on_or_between(self, start, end):
         return (self >= start) & (self <= end)
 
+    def if_null_then(self, other):
+        return case(
+            when(self.is_not_null()).then(self),
+            default=parse_date_if_str(other),
+        )
```

Comment:
Good catch!

Reply:
It was only when I rewrote the test to use introspection rather than a static list of functions that I caught this!